.github/workflows/apps/outdated_github_majors.go

```diff
@@ -0,0 +1,245 @@
+		if err != nil {
+			return err
+		}
+
+		// Check if the path matches the contrib repository pattern
+		if !info.IsDir() || !repoRegex.MatchString(path) {
+			return nil
+		}
+
+		// Look for go.mod file
+		goModPath := filepath.Join(path, "go.mod")
+		if _, err := os.Stat(goModPath); err != nil {
+			return nil // No go.mod file, skip
+		}
+
+		// Parse the go.mod file to find the version
+		version, err := extractVersionFromGoMod(goModPath, repository)
+		if err != nil {
+			return nil
+		}
+
+		// Update the latest version if it's greater
+		if latestVersion == "" || semver.Compare(version, latestVersion) > 0 {
+			latestVersion = version
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("error walking contrib directory: %v", err)
+	}
+
+	if latestVersion == "" {
+		return "", errors.New("no matching contrib repository with a valid version found")
+	}
+
+	return latestVersion, nil
+}
+
+func extractVersionFromGoMod(goModPath string, repository string) (string, error) {
+	data, err := os.ReadFile(goModPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read go.mod: %v", err)
+	}
+	var versions []string
+
+	// parse go.mod
+	modFile, err := modfile.Parse(goModPath, data, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse go.mod: %v", err)
+	}
+
+	for _, req := range modFile.Require {
+		if strings.Contains(req.Mod.Path, repository) { // check if module path contains repository substring
+			if semver.IsValid(req.Mod.Version) {
+				versions = append(versions, req.Mod.Version)
+			}
+		}
+
+	}
+
+	if len(versions) == 0 {
+		return "", errors.New("no valid versions found for repository in go.mod")
+	}
+
+	// Sort versions in descending order
+	// Return the greatest version
+	sort.Slice(versions, func(i, j int) bool {
+		return semver.Compare(versions[i], versions[j]) > 0
+	})
+
+	return versions[0], nil
+
+}
+
+// ValidateRepository checks if the repository string starts with "github.com" and ends with a version suffix.
+// Returns true if the repository is valid, false otherwise.
+func ValidateRepository(repo string) bool {
+	const prefix = "github.com/"
+	if !strings.HasPrefix(repo, prefix) {
+		return false
+	}
+
+	lastSlashIndex := strings.LastIndex(repo, "/")
+	if lastSlashIndex == -1 || !strings.HasPrefix(repo[lastSlashIndex:], "/v") {
+		return false
+	}
+
+	return true
+}
+
+func main() {
+
+	packages := instrumentation.GetPackages()
+	integrations := make(map[string]struct{})
+
+	for _, info := range packages {
+		repository := info.TracedPackage
+
+		// repo starts with github and ends in version suffix
+		if ValidateRepository(repository) {
+			const prefix = "github.com"
+
+			// check if we've seen this integration before
+			lastSlashIndex := strings.LastIndex(repository, "/")
+			baseRepo := repository[len(prefix)+1 : lastSlashIndex]
+			if _, exists := integrations[baseRepo]; exists {
+				continue
+			}
+
+			fmt.Printf("Base repo: %s\n", baseRepo)
```

Comment:
nit: if we can use `log` instead of `fmt` it would be cleaner and allow to parse a bit the output.